### PR TITLE
Multi-purpose homepage.

### PR DESCRIPTION
**Pull request by lcrilly**
Multi-purpose homepage.

This pull request repurposes the index (/) page to become a conventional, multi-purpose homepage, serving content from multiple sources.

The new-style homepage has three sections:

1. Introductory text about nginx
2. Latest news
3. Other NGINX projects

As a precursor, the existing news content is moved to a dedicated news page.

Commits:

- **Dedicated news page.**
- **Multi-purpose index page.**
- **Regenerated.**

### CODE REVIEW

File: xml/homepage.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+and mail proxy server.
+Originally written by <link url="http://sysoev.ru/en/">Igor Sysoev</link>
+and distributed under the
+<link url="../LICENSE">2-clause BSD License</link>.
+Enterprise distributions, commercial support and training are
+<link url="enterprise.html">available from F5, Inc.</link>
+</para>
+
+<para>
+<link url="en/docs/">Docs</link>
```
Comment (y82) [inline xml/homepage.xml]: `<link doc="en/docs/index.xml">Docs</link>`

File: xml/homepage.xml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+•
+<link url="https://github.com/nginx/nginx">Code</link>
+•
+<link url="en/linux_packages.html">Linux Packages</link>
```
Comment (y82) [inline xml/homepage.xml]: `<link doc="en/linux_packages.xml">Linux Packages</link>`

File: xml/homepage.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+•
+<link url="en/docs/beginners_guide.html">Beginner's Guide</link>
```
Comment (y82) [inline xml/homepage.xml]: `<link doc="en/docs/beginners_guide.xml">Beginner's Guide</link>`

**Review by y82 — COMMENTED** [review]
I also added the patch with these fixes (it can be merged or just deleted if the feedback is fixed locally)
I didn't check the code part, however I checked build procedure, final on the local machine, in docker, and from the remote server.

**Comment by pluknet** [comment]
Rendering issue 1: years are glued together

Screenshot 2024-10-03 at 18 08 15

Rendering issue 2: broken layout

Screenshot 2024-10-03 at 18 08 38

Rendering issue 3: holes in the menu

Screenshot 2024-10-03 at 18 09 09

**Review by pluknet — CHANGES_REQUESTED** [review]
Need to sort out why layout is broken (see earlier posted screenshots)

**Comment by lcrilly** [comment]
> Need to sort out why layout is broken (see earlier posted screenshots)

This is a CSS fail. Check the generated `<head>` block. @y82 comments that the site build was successful.

**Comment by pluknet** [comment]
Overall, looks good enough.

**Comment by pluknet** [comment]
OTOH, the horizontal menu looks weird IMHO, and also it partially duplication the existing vertical menu.
I'd rather remove it.

**Comment by pluknet** [comment]
Rebased on top off #84

**Comment by pluknet** [comment]
Removed n_hex.svg (approved by @lcrilly in private).

File: xml/homepage.xml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+Enterprise distributions, commercial support and training are
+<link url="enterprise.html">available from F5, Inc.</link>
+</para>
+
+<para>
+<link doc="en/docs/index.xml">Docs</link>
+•
+<link url="https://github.com/nginx/nginx">Code</link>
+•
+<link doc="en/linux_packages.xml">Linux Packages</link>
```
[the remark this answers is not in the document]
Comment (lcrilly) [reply xml/homepage.xml:31]: Good idea, especially as the Install page links to the packages. Will update this with link to the .xml (not .html)

**Comment by pluknet** [comment]
Incorporated feedback into to 3 original commits.

File: xml/homepage.xml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+<para>
+nginx ("<i>engine x</i>") is an HTTP web server, reverse proxy,
+content cache, load balancer,
+TCP/UDP proxy server,
+and mail proxy server.
+Originally written by <link url="http://sysoev.ru/en/">Igor Sysoev</link>
+and distributed under the
+<link url="../LICENSE">2-clause BSD License</link>.
+Enterprise distributions, commercial support and training are
+<link url="enterprise.html">available from F5, Inc.</link>
```
Comment (kleisauke) [inline xml/homepage.xml:23]: https://nginx.org/enterprise.html is 404 for me.

Comment (lcrilly) [reply xml/homepage.xml:23]: Thank you! Submitted fix as https://github.com/nginx/nginx.org/pull/86

Comment (pluknet) [reply xml/homepage.xml:23]: Fixed, thanks. The update should appear on the site shortly.

**Pull request merged**